### PR TITLE
Example with element_proc option

### PR DESCRIPTION
**Pull request by wooly**

### CODE REVIEW

File: app/components/view_component/form/collection_check_boxes_component.rb [hunk 1]
```diff
@@ -26,6 +26,8 @@ def initialize( # rubocop:disable Metrics/ParameterLists
       end
 
       def call # rubocop:disable Metrics/MethodLength
+        build_proc = options.delete(:build_proc)
```
Comment (Spone) [inline app/components/view_component/form/collection_check_boxes_component.rb:29]:
```suggestion
        element_proc = options.delete(:element_proc)
```

File: app/components/view_component/form/collection_check_boxes_component.rb [hunk 2]
```diff
@@ -36,7 +38,7 @@ def call # rubocop:disable Metrics/MethodLength
           options,
           html_options,
           &content
-        ).render
+        ).render(&build_proc)
```
Comment (Spone) [inline app/components/view_component/form/collection_check_boxes_component.rb:41]:
```suggestion
        ).render(&element_proc)
```

File: spec/view_component/form/collection_check_boxes_component_spec.rb [hunk 3]
```diff
@@ -53,6 +53,27 @@
     end
   end
 
+  context "with a build proc" do
+    let(:options) do
+      {
+        build_proc: proc do |b|
```
Comment (Spone) [inline spec/view_component/form/collection_check_boxes_component_spec.rb:59]:
```suggestion
        element_proc: proc do |b|
```

**Review by Spone — CHANGES_REQUESTED** [review]
We suggest a small naming change.

Also, can you implement the same changes in `CollectionRadioButtonsComponent`?

**Comment by Spone** [comment]
We'll take care of updating the README.

**Comment by wooly** [comment]
> We suggest a small naming change.
> 
> Also, can you implement the same changes in `CollectionRadioButtonsComponent`?

Yep, can do that in a few hours :+1:

**Comment by wooly** [comment]
@Spone how does that look?

**Comment by Spone** [comment]
@wooly looking good! We'll update the README and merge.

**Review by Spone — APPROVED** [review]

**Comment by wooly** [comment]
Many thanks!

**Comment by rubiii** [comment]
Hey. We started using this library a couple of weeks ago and this feature would be greatly appreciated.
So thank you very much!

We would suggest a small change though. Currently, the `element_proc` is expected to be passed via
`options` and that's fine, as long as you don't have any `html_options`. Since both args are optional,
we fixed this on our side by just looking into both hashes:

```ruby
element_proc = options.delete(:element_proc) || html_options.delete(:element_proc)
```

What do you think?

**Comment by Spone** [comment]
Good call @rubiii! Can you make the change @wooly please?

**Comment by rubiii** [comment]
@wooly I could prepare a PR to your branch if that would help you out.
Given my proposed changed looks valid to you of course.

**Comment by wooly** [comment]
@rubliii that would be super if you could do that as I'm currently on holiday! Thanks! I guess what we're saying then is that the element_proc can be passed in via both options and html_options? Should we then delete from both in the extremely unlikely case that it's specified in both?

**Comment by rubiii** [comment]
Thanks for replying from your holidays @wooly :)

Good question. We could check and delete the option from both hashes and also raise an error to cover the probably unlikely case where both option hashes include an `element_proc`. That would be a little more code, but it's probably good to prevent any confusion.

**Comment by Spone** [comment]
> That would be a little more code, but it's probably good to prevent any confusion.

Agreed!

**Comment by Spone** [comment]
@rubiii @wooly let us know if you need help to wrap this up!

**Comment by wooly** [comment]
Sorry for the delay. Have merged @rubiii's PR into my branch now, so we should be good to go.

**Comment by Spone** [comment]
Hmm we have to fix the CI failures on `main` first :/

**Comment by wooly** [comment]
@Spone anything I can do to get this over the line?

**Comment by Spone** [comment]
Hi @wooly thanks for your time on this PR. The past few months have been pretty hectic at work and now I'm on holiday to recharge batteries 🙂

I think we still have to figure out why CI is failing on main before releasing a new version, but in the meantime I think we can go ahead and merge this, if you can update the changelog beforehand it would be perfect 🙏

**Comment by wooly** [comment]
Thanks @Spone - have updated CHANGELOG.md

**Pull request merged**